azure-kusto-ingest/example.js

```diff
@@ -132,34 +58,80 @@ const streamingIngestClient = new StreamingIngestClient(
     props2
 );
 
-// Ingest from file with either file path or FileDescriptor
-streamingIngestClient.ingestFromFile("file.json", null, (err) => {
-    if (err) {
+startStreamingIngestion();
+
+async function startIngestion() {
+    console.log("Ingest from file");
+    try {
+        await ingestClient.ingestFromFile("file.json");
+        console.log("Ingestion done?");
+
+        await waitForStatus();
+    }
+    catch (err) {
         console.log(err);
     }
 
-    console.log("Ingestion done");
-});
-
-// Ingest from stream with either ReadStream or StreamDescriptor
-const stream = fs.createReadStream("file.json");
+    try {
+        await ingestClient.ingestFromBlob(new BlobDescriptor("https://<account>.blob.core.windows.net/<container>/file.json.gz", 1024 * 50 /* 50MB file */));
+        console.log("Ingestion done?");
 
-streamingIngestClient.ingestFromStream(stream, null, (err) => {
-    if (err) {
+        await waitForStatus();
+    }
+    catch (err) {
         console.log(err);
     }
+}
 
-    console.log("Ingestion done");
-});
+async function waitForStatus(numberOFIngestions = 1) {
+    while (await statusQueues.failure.isEmpty() && await statusQueues.success.isEmpty()) {
+        console.log("Waiting for status...");
+        await sleep(1000);
```

Comment:
Why 1 sec? Can we wait for a condition instead? My concern is that this case might cause sporadic failures

Reply:
It's an example code...

Reply:
I will clarify. This is an example but keep in mind this is our customer way of learning. So they might also add the wait for 1 sec. In their real code it might work for sometime, and something it might fail. It is always a good practice to avoid time based waiting and move to condition base waiting

Reply:
With this I agree, but any ideas for a condition a client can wait on to check ingestion status?

Reply:
we have a condition in the while... what do you suggest to do instead?

Reply:
Selenium solved this in a very elegant way. Their way of solving such case was with such API:
wait.Until(statusQueues => statusQueues.failure.isEmpty() && await statusQueues.success.isEmpty());

You can see how they implemented it in here: https://github.com/SeleniumHQ/selenium/blob/2a5b016fa928e20c1f00b404a8f0d63d4b5d5f80/dotnet/src/webdriver/Support/DefaultWait%7BT%7D.cs

If needed let's take it offline

Reply:
I think that for an example the current impl is enough